software/contrib/euclidean_rhythm.py

```diff
@@ -0,0 +1,356 @@
+                    if len(self.notes_list) < 10:
+                        oled.fill_rect(box_position[self.position][0]+2+i*4, box_position[self.position][1]+8, 3, box_size[1]-8, 1)
+                        oled.fill_rect(box_position[self.position][0]+2+i*4, box_position[self.position][1]+5, 3, 2, 1)
+                    elif len(self.notes_list) < 14:
+                        oled.fill_rect(box_position[self.position][0]+2+i*3, box_position[self.position][1]+8, 2, box_size[1]-8, 1)
+                        oled.fill_rect(box_position[self.position][0]+2+i*3, box_position[self.position][1]+5, 2, 2, 1)
+                    elif len(self.notes_list) < 20:
+                        oled.fill_rect(box_position[self.position][0]+2+i*2, box_position[self.position][1]+8, 1, box_size[1]-8, 1)
+                        oled.fill_rect(box_position[self.position][0]+2+i*2, box_position[self.position][1]+5, 1, 2, 1)
+                # Low
+                else:
+                    if len(self.notes_list) < 10:
+                        oled.fill_rect(box_position[self.position][0]+2+i*4, box_position[self.position][1]+15, 3, box_size[1]-15, 1)
+                        oled.fill_rect(box_position[self.position][0]+2+i*4, box_position[self.position][1]+12, 3, 2, 1)
+                    elif len(self.notes_list) < 14:
+                        oled.fill_rect(box_position[self.position][0]+2+i*3, box_position[self.position][1]+15, 2, box_size[1]-15, 1)
+                        oled.fill_rect(box_position[self.position][0]+2+i*3, box_position[self.position][1]+12, 2, 2, 1)
+                    elif len(self.notes_list) < 20:
+                        oled.fill_rect(box_position[self.position][0]+2+i*2, box_position[self.position][1]+15, 1, box_size[1]-15, 1)
+                        oled.fill_rect(box_position[self.position][0]+2+i*2, box_position[self.position][1]+12, 1, 2, 1)
+
+class EuclideanRhythm(EuroPiScript):
+    def __init__(self):
+        # length, fill, offset, cv
+        self.notes_para = [[1, 0, 0, 0], [1, 0, 0, 0], [1, 0, 0, 0],
+                           [1, 0, 0, 0], [1, 0, 0, 0], [1, 0, 0, 0]]
+        
+        # length, fill, offset
+        self.notes_para_final = [[1, 0, 0], [1, 0, 0], [1, 0, 0],
+                                 [1, 0, 0], [1, 0, 0], [1, 0, 0]]
+
+        
+        self.selection_idx = 0 # change editing gates
+        self.editing_idx = 0 # change editing parameters
+        self.blink_flg = 0 # blink selection with tempo
+        self.clock_flg = 0 # 0: internal clock, 1: external clock (from digital in)
+        self.steps_idx = [0 for i in range(6)] # step of each sequencer
+        self.tempo = 60 # current tempo
+        self.deadline = 0  # internal clock parameter
+        self.prev_clock = 0 # external clock parameter
+        self.cv_value = 0
+
+        @b1.handler_falling
+        def para_sub():
+            if self.selection_idx <= 5:
+                if self.editing_idx == 0:
```

Comment:
I think that this entire `if`/`elif` block could be replaced with:

`self.notes_para[self.selection_idx][self.editing_idx] -= 1`

The same change follows in the 'add' section below.

Reply:
Changed